### ISSUE TITLE
Curve editor missing X range inputs

### ISSUE THREAD

**Issue by krotter0** [Facepunch/sbox-issues#5265]
### For?

S&Box

### What can't you do?

The curve editor is missing input fields for the X range, even though it's something that curves can handle. It'd be nice to be able to change it.

### How would you like it to work?

![img](https://github.com/Facepunch/sbox-issues/assets/65136069/c235d7e8-96ed-4799-baf9-6ac162b53758)

### What have you tried?

Editing the editor code in base.

### Additional context

I'm pretty sure this wasn't missing a while ago, since I have some old projects which rely on the x range of curves.

**Comment by aylaylay** [comment]
Probably one for @garrynewman I'm not sure why the properties for time range is missing, probably because curve editor is mostly used for particles and the time range is usually 0-1?

**Comment by garrynewman** [comment]
Yeah normalized time is on purpose. That isn't really what this is used for.. and although it could be adapted, we'd need a really good reason to do it

**Comment by krotter0** [comment]
One of my main uses was for a vehicle controller, where the x axis was used for speed, rpm or any other parameter and y for some force output:
![cttorq](https://github.com/Facepunch/sbox-issues/assets/65136069/92aba734-dfc3-41e4-bab4-2c7db9f86e24)
![ctgrcrvs](https://github.com/Facepunch/sbox-issues/assets/65136069/cfa1a885-60c1-45aa-a611-68777e2dbed2)
_[(Source: Car Physics for Games by Marco Monster)](https://www.asawicki.info/Mirror/Car%20Physics%20for%20Games/Car%20Physics%20for%20Games.html)_

More recently, I was missing this when I was making a physics player controller, where I wanted to lower the player's acceleration when they already had a lot of speed. A curve allowed me to quicky tweak this setting in the inspector, but I had to introduce a vector2 property which is used just for remapping the player's speed to a value between 0-1. I had to change two different values for the same thing.

I also made a player sound component for this same project, which uses curves similarly - they take the player's speed for x and output the sound's pitch/volume on y.
